website_docs/_index.md

```diff
@@ -30,3 +30,162 @@ The current release can be found [here](https://github.com/open-telemetry/opente
+
+  # ... and after creation
+  span.set_attribute("animals", ["elephant", "tiger"])
+
+  span.add_attributes({ "my.cool.attribute" => "a value", "my.first.name" => "Oscar" })
+end
+```
+
+#### Semantic Attributes
+
+Semantic Attributes are attributes that are defined by the OpenTelemetry Specification in order to provide a shared set of attribute keys across multiple languages, frameworks, and runtimes for common concepts like HTTP methods, status codes, user agents, and more. These attributes are available in the [Semantic Conventions gem](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/semantic_conventions).
+
+Tracing semantic conventions can be found in [this document](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions)
+
+#### Events
+
+An event is a human-readable message on a span that represents "something happening" during it's lifetime. For example, imagine a function that requires exclusive access to a resource that is under a mutex. An event could be created at two points - once, when we try to gain access to the resource, and another when we acquire the mutex.
+
+```ruby
+span.add_event("Acquiring lock")
+if mutex.try_lock
+  span.add_event("Got lock, doing work...")
+  # some code here
+  span.add_event("Releasing lock")
+else
+  span.add_event("Lock already in use")
+end
+```
+
+A useful characteristic of events is that their timestamps are displayed as offsets from the beginning of the span, allowing you to easily see how much time elapsed between them.
+
+Events can also have attributes of their own e.g.
+
+```ruby
+span.add_event("Cancelled wait due to external signal", attributes: { "pid" => 4328, "signal" => "SIGHUP" })
+```
+
+### Propagators and Context
+
+Traces can extend beyond a single process. This requires _context propagation_, a mechanism where identifiers for a trace are sent to remote processes.
+
+In order to propagate trace context over the wire, a propagator must be registered with the OpenTelemetry SDK.
+The W3 TraceContext and Baggage propagators are configured by default, however operators may override this value by setting `OTEL_PROPAGATORS` environment variable to a comma separated list of [propagators](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/propagator).
+
+```sh
+export OTEL_PROPAGATORS=tracecontext,baggage,b3
+```
+
+You will also have to add a gem dependency:
+
+```sh
+gem install opentelemetry-propagator-b3
+```
+
+or via `Bundler` by adding the following to your `Gemfile`:
+
+```ruby
+gem 'opentelemetry-propagator-b3'
+```
+
+followed by:
+
+```sh
+bundle install
+```
+
+### Automatic Instrumentation
+
+Automatic instrumentation, broadly, refers to instrumentation code that you didn't write. OpenTelemetry Ruby supports this process through prepended modules and helper functions around many popular frameworks and libraries.
```

Comment:
Technically, it refers to instrumentation code that you didn't write and that the authors of the instrumented code didn't write either 😄 .

Along those lines, we ought to include guidelines for authors of popular frameworks and libraries that may wish to instrument their code directly. The key message is that frameworks and libraries should depend only on the API (and semantic conventions), not the SDK, resources or exporters.